apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletApiAdvice.java

```diff
@@ -178,20 +160,12 @@ public static Object onEnterServletService(@Advice.Argument(0) ServletRequest se
         return ret;
     }
 
-    private static boolean areNotEqual(@Nullable Object first, @Nullable Object second) {
-        if (first == null) {
-            return second != null;
-        } else {
-            return !first.equals(second);
-        }
-    }
-
-    @Advice.OnMethodExit(suppress = Throwable.class, onThrowable = Throwable.class, inline = false)
-    public static void onExitServletService(@Advice.Argument(0) ServletRequest servletRequest,
-                                            @Advice.Argument(1) ServletResponse servletResponse,
-                                            @Advice.Enter @Nullable Object transactionOrScopeOrSpan,
-                                            @Advice.Thrown @Nullable Throwable t,
-                                            @Advice.This Object thiz) {
+    public static <ServletRequest, ServletResponse, HttpServletRequest, HttpServletResponse, ServletContext> void onExitServlet(ServletRequest servletRequest,
```

Comment:
The generic type definitions using classnames are confusing, and would be even more confusing if any of these were imported to be used somewhere in a later change. I suggest changing them to uppercase, eg ServletRequest -> REQUEST, HttpServletRequest-> HTTPREQUEST etc

Reply:
fixed